### ISSUE TITLE
[BUG] Improved Devouring Plague Talent uses wrong haste

### ISSUE THREAD

**Issue by InDebt** [JamminL/cata-classic-bugs#971]
#### 1.1. Description
* The talent [Improved Devouring Plague](https://www.wowhead.com/cata/spell=63626/improved-devouring-plague)'s damage calculation is using strange haste values. Changing the player's haste value effects how many ticks the instant damage is using without even passing a breakpoint for DoT ticks.

| Haste Total | DP Tick Count | DP Tick Damage | Imp DP Damage |
| --- | --- | --- | --- |
| 21.01%  | 10 | 506 | 1366 |
| 22.13%  | 10 | 506 | 1517 |

So cast the with $`21.01\%`$ haste uses only 9 ticks to calculate the total damage: $` 9 * 506 * 0.3 = 1,366.2`$  
While the other cast uses the correct 10 ticks: $` 10 * 506 * 0.3 = 1,518`$

  #### 1.2. How to Reproduce
1. Create a new level 80 priest template character
2. Spec into 3/3 Darkness, 2/2 Imp Devouring Plague and 1/1 Shadowform
3. Change your starter gear from your bags to get to ~21% haste
4. Cast DP
5. Equip more haste to get > 22% haste
6. Caste DP again

  #### 1.3  Source Material
* Ingame tests

#### 2.1. Description

Devouring Plague has its haste breakpoints for 10 / 11 Ticks at ~18.52% and 31.12% total haste according to the shadow priest guide. This can be confirmed to now align with ingame behavior (at least for 10 ticks - don't have enough haste for 11) with the latest DoT Haste fixes. Imp DP should deal 30% of the total damage dealt.

  #### 2.2 Source Material
* [Video of damage aligning](https://www.youtube.com/watch?v=hbmtLM3xBsI)
* [Haste breakpoints for priest spells](https://web.archive.org/web/20120619131502/https://elitistjerks.com/f77/t124358-shadow_priest_pve_guide_4_3_updated/)

**Comment by Polynomix** [comment]
Don't you need 21.32% haste to go from 10 to 11 ticks with shadowform and darkness ? The bug i see here is that you dont get an extra tick on the dot at 22.13% haste.

(Base_Tick_Speed / ((Base_Frequency * Base_Tick_Speed) / (No_of_Ticks - 0.5) + 0.00049999)) / shadowform / darkness
(3 / ((8 * 3) / (11 - 0.5) + 0.00049999)) / 1.05 / 1.03 = ~1.2133

which is 0.2133*3279 = 700 haste ratings at level 80

**Comment by InDebt** [comment]
Well the 21.33% haste would be the haste from gear as you already divided by 5% from Shadowform and 3% from Darkness. The character screen already applies the haste from both in the display. So you need 21.33 haste from gear and 31.21% total (in char screen) 🤔  or am I just crazy now

**Comment by Polynomix** [comment]
you're right i dont know why, i jsut assumed your haste value where based one

**Comment by TheHate123** [comment]
Kinda hard to actually test this with green gear on the beta. But how it's supposed to work is described here in a comment from Kilee, the moderator on how2priest -> https://web.archive.org/web/20121022043614/http://howtopriest.com/viewtopic.php?f=50&t=1194

tl;dr it should only take your haste from gear, and completely ignore all buffs, racials and alike. You get the extra damage on Improved Devouring Plague at 1601 and 3202 Haste from gear only exactly, regardless of any talents or Dark Intent or whatever.

**Comment by InDebt** [comment]
Well that's some intersting source. This would make this intended behavior.

3202 / 1601 haste would make for 25% and 12.5% breakpoints. Which would be roughly 411 haste rating from gear on levle 80 for the first breakpoint (10 ticks). Which would be $`1.125 * 1.03 * 1.05 = 1,21668`$ which would exactly line up with the observed breakpoint I initially reported.

It seems Imp:DP is strictly rounding up, so as soon as 9.01 ticks fit into the base duration it calculates damage for 10.
